File: claw_crane_game.py
def solution(board, moves):
    answer = 0
    stack = []
    
    for move in moves:
        for row, _ in enumerate(board):
            if board[row][move - 1] != 0:
                item = board[row][move - 1]
                board[row][move - 1] = 0
                if len(stack) == 0 or stack[-1] != item:
                    stack.append(item)
                else:
                    stack.pop()
                    answer += 2
                break
    
    return answer

File: claw_crane_game.cpp
#include <vector>
#include <stack>
using namespace std;

stack<int> st;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0, move, item;
    int boardSize = board[0].size();
    
    for(int i=0; i<moves.size(); i++) {
        move = moves[i] - 1;
        for(int row=0; row<boardSize; row++) {
            if(board[row][move] != 0) {
                item = board[row][move];
                board[row][move] = 0;
                if(st.empty() || item != st.top())
                    st.push(item);
                else if(item == st.top()) {
                    st.pop();
                    answer += 2;
                }
                break;
            }
        }
    }
    
    return answer;
}